src/Apps/Order2/Routes/Checkout/Components/Order2PricingBreakdown.tsx

```diff
@@ -0,0 +1,133 @@
+import { Flex, Spacer, Text } from "@artsy/palette"
+import type {
+  Order2PricingBreakdown_order$data,
+  Order2PricingBreakdown_order$key,
+} from "__generated__/Order2PricingBreakdown_order.graphql"
+import { graphql, useFragment } from "react-relay"
+
+interface Order2PricingBreakdownProps {
+  order: Order2PricingBreakdown_order$key
+}
+
+const knownLineTypes = [
+  "ShippingLine",
+  "TaxLine",
+  "SubtotalLine",
+  "TotalLine",
+] as const
+
+type KnownLineType = Extract<
+  Order2PricingBreakdown_order$data["pricingBreakdownLines"][number],
+  { __typename: "ShippingLine" | "TaxLine" | "SubtotalLine" | "TotalLine" }
+>
+
+const isKnownLineType = (
+  line: Order2PricingBreakdown_order$data["pricingBreakdownLines"][number],
+): line is KnownLineType => {
+  return !!line && knownLineTypes.includes(line.__typename as any)
+}
+
+export const Order2PricingBreakdown: React.FC<Order2PricingBreakdownProps> = ({
+  order,
+}) => {
+  const { pricingBreakdownLines } = useFragment(FRAGMENT, order)
+  return (
+    <>
+      {pricingBreakdownLines.map((line, index) => {
+        if (!(line && isKnownLineType(line))) {
+          return null
+        }
+
+        const typename = line.__typename
+        let variant: "sm" | "sm-display" = "sm"
+        let fontWeight = "regular"
+        let color = "mono60"
+        let withAsterisk = false
+
+        let amountText: string
+        switch (typename) {
+          case "SubtotalLine":
+            amountText = line.amount
+              ? `${line.amount.currencySymbol} ${line.amount.amount}`
+              : ""
+            break
+          case "ShippingLine":
+            amountText = line.amount
+              ? `${line.amount.currencySymbol} ${line.amount.amount}`
+              : (line.amountFallbackText ?? "")
+            break
+          case "TaxLine":
+            amountText = line.amount
+              ? `${line.amount.currencySymbol} ${line.amount.amount}`
+              : (line.amountFallbackText ?? "")
+            withAsterisk = true
+            break
+          case "TotalLine":
+            variant = "sm-display"
+            fontWeight = "medium"
+            color = "mono100"
+            amountText = (line.amount?.display || line.amountFallbackText) ?? ""
```

Comment:
Technically we should never actually wind up with an empty string for any of these values - there should be either an amount or a fallback text.

Reply:
If that is explicitly the case, then why add the additional ?? "". Optimizing for a case that will never happen

Reply:
Why are we guarding against cases that can never actually execute?

Reply:
Type-wise both of these things are nullable in the schema. What is the preferred way to handle it? `!`?

Reply:
If its to satisfy the type-checker can do one of two things:
- add `@required(action: NONE)` on the field in the query
- cast `as string` on the line